### PR TITLE
import changes from pay-server

### PR DESCRIPTION
**Pull request by ptarjan**
I wasn't on a few of the changes in pay-server so I'm not sure if I did this right. These are diverging pretty hard now. I'm going to start advocating to make two PRs so I stop doing this syncing.

### Test plan

See included automated tests.

### CODE REVIEW

File: gems/sorbet-runtime/test/types/configuration.rb [hunk 1]
```diff
@@ -59,7 +60,7 @@ def self.receive(*); end
     describe 'sig_builder_error_handler' do
       describe 'when in default state' do
         it 'raises an error' do
-          @mod.sig {returns(Symbol).void}
+          @mod.sig {generated.returns(Symbol).checked(:always)}
```
Comment (jez) [inline gems/sorbet-runtime/test/types/configuration.rb]: @ptarjan every change in this file is a revert of the change to disable generated / checked for open source.

File: gems/sorbet-runtime/test/types/configuration.rb [hunk 2]
```diff
@@ -6,7 +6,8 @@ class ConfigurationTest < Critic::Unit::UnitTest
     before do
       @mod = Module.new do
         extend T::Sig
-        # Make it public for testing only public_class_method :sig
+        # Make it public for testing only
+        public_class_method :sig
```
Comment (jez) [inline gems/sorbet-runtime/test/types/configuration.rb]: actually jk this change is ok to keep

File: gems/sorbet-runtime/test/types/configuration.rb (hunk 1 above)
Comment (ptarjan) [reply gems/sorbet-runtime/test/types/configuration.rb]: Yeah, I'm having a bunch of trouble keeping track of which side is the source of truth for any given change. I'll revert most of this file, but we have to stop this lunacy

Comment (jez) [reply gems/sorbet-runtime/test/types/configuration.rb]: strong +1

**Comment by ptarjan** [comment]
@jez what's the plan for `.checked`? Should I delete calls to it in `sorbet-runtime`?

**Comment by jez** [comment]
> @jez what's the plan for `.checked`? Should I delete calls to it in `sorbet-runtime`?

If they want to be using `.checked(:never)` I think they should instead use `T::Sig::WithoutRuntime.sig`.

**Comment by jez** [comment]
I synced these in other PRs.

**Pull request closed without merging**